# Based on Avian (https://github.com/ReadyTalk/avian).
# Original work Copyright (c) 2008-2015, Avian Contributors.
# Original work licensed under ISC; see LICENSE.txt:
# https://github.com/BasaltProject/Shim/blob/main/LICENSE.txt
#
# Modifications Copyright (c) 2026 BasaltMC.
# Modifications licensed under MIT; see LICENSE:
# https://github.com/BasaltProject/Shim/blob/main/LICENSE
#
# SPDX-License-Identifier: ISC AND MIT

--- openjdk/Inet4AddressImpl.c
+++ openjdk/Inet4AddressImpl.c
@@ -461,6 +461,21 @@
     return JNI_FALSE;
 }

+#ifdef WIN32
+DWORD WINAPI IcmpSendEcho2Ex(HANDLE,
+			     HANDLE,
+			     LPVOID,
+			     PVOID,
+			     IPAddr,
+			     IPAddr,
+			     LPVOID,
+			     WORD,
+			     LPVOID,
+			     LPVOID,
+			     DWORD,
+			     DWORD);
+#endif
+
 /**
  * ping implementation.
  * Send a ICMP_ECHO_REQUEST packet every second until either the timeout
